### DISCUSSION TITLE
Midi to keyboard translator

### DISCUSSION THREAD

**Discussion by Didkrak** [Monika-After-Story/MonikaModDev#7840] [general - submod / external content]
So this isn't exactly a submod, but figured some others might like it. I made a midi-event to keyboard-press translator so we can control Monika's piano with a connected midi piano, the repository is here: [https://github.com/Didkrak/Midi-Keyboard](https://github.com/Didkrak/Midi-Keyboard). It only works on windows unfortunately, I could not find any cross-platform methods that work.

Other, more complete, translators already exist, but all the ones I tested seem to use Virtual Keyboard signals which pygame (and thus MAS) does not register as a keypress. This one uses scancodes instead.

I'll likely eventually make a submod to extend Monika's piano so we can use the full range of a connected piano, but for now this works. It's only been tested on two computers and one midi device though, so if you run into problems let me know.

**Comment by ibm2431** [comment]
**Thank you!** I've been looking for a solution to this for a while now!

Will confirm that the various existing MIDI-to-key programs did not work. They'd register in notepad, but not in MAS.

But this works in MAS for my Yamaha P115!

(Note: Need to have the device powered on when starting to avoid pygame throwing out an exception for unhandled device ID.)

I was even able to build this from source into an .exe myself, too!
